pinot-core/src/main/java/org/apache/pinot/core/query/aggregation/function/BaseBooleanAggregateFunction.java

```diff
@@ -0,0 +1,176 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.pinot.core.query.aggregation.function;
+
+import java.util.Map;
+import java.util.function.IntPredicate;
+import org.apache.pinot.common.request.context.ExpressionContext;
+import org.apache.pinot.common.utils.DataSchema;
+import org.apache.pinot.core.common.BlockValSet;
+import org.apache.pinot.core.query.aggregation.AggregationResultHolder;
+import org.apache.pinot.core.query.aggregation.ObjectAggregationResultHolder;
+import org.apache.pinot.core.query.aggregation.groupby.GroupByResultHolder;
+import org.apache.pinot.core.query.aggregation.groupby.ObjectGroupByResultHolder;
+import org.apache.pinot.spi.data.FieldSpec;
+import org.roaringbitmap.RoaringBitmap;
+
+
+public abstract class BaseBooleanAggregateFunction extends BaseSingleInputAggregationFunction<Integer, Integer> {
+
+  private final BooleanMerge _merger;
+  private final boolean _nullHandlingEnabled;
+
+  protected enum BooleanMerge {
+    AND {
+      @Override
+      int merge(Integer left, int right) {
```

Comment:
Discussed offline, we'll keep the enum to avoid duplicating code - I wrote this JMH to confirm that there shouldn't be a performance impact (see below). We can always go back and optimize it if need be.

```
  @Setup
  public void setUp() {
    Random random = new Random();
    _ints = IntStream.generate(random::nextInt).limit(1000).toArray();
    _enum = MyEnum.ADD;
  }

  @Benchmark
  public int inline() {
    int sum = 0;
    for (int i : _ints) {
      sum = sum + i;
    }
    return sum;
  }

  @Benchmark
  public int delegate() {
    int sum = 0;
    for (int i : _ints) {
      sum = _enum.apply(sum, i);
    }
    return sum;
  }
```

The result was:
```
Benchmark                       Mode  Cnt  Score   Error  Units
BenchmarkEnumOverhead.delegate  avgt    5  0.306 ± 0.040  us/op
BenchmarkEnumOverhead.inline    avgt    5  0.306 ± 0.033  us/op
```